Update ILP Glossary (#123)
interledger/interledger.github.io: docs/markdown/glossary.md

# Glossary

This glossary provides definitions for key terms in the Interledger ecosystem.

## Account
   

> An accounting relationship between two Interledger participants. In Interledger, two participants establish an account with one another to track the current obligations they hold with one another. An Interledger account has a [balance](#balance) denominated in a mutually agreed upon asset (e.g. USD) at an agreed upon scale (e.g. 2).                                                                        

Reviewer: Since "scale" is not defined here (and AFAIK is a concept we invented), maybe replace "scale" with "denomination"?

## Balance 

An account balance is the net total of the amounts on any packets "successfully" routed between the peers. When a [node](#node) sends an ILP [Prepare packet](#interledger-packet) to a [peer](#peer), and receives a valid ILP [Fulfill](#interledger-packet) response, before the expiry of the ILP Prepare, their balance with that peer decreases, meaning the nodes owes the peer money. When a node receives an ILP Prepare packet and returns a valid ILP Fulfill response, before the expiry of the ILP Prepare, their balance with that peer increases, meaning the peer owes the node money.                              

## Bandwidth

The limit, set by a [router](#router), of the total value an account can have pending, or "in-flight" at any given moment in time. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's liquidity with its peers.                                                                       

## Clearing    

> The transferring (transmitting, reconciling, and confirming) of funds from payer to payee, such that the payer's account is debited for the promised amount and the payee's account is credited for the equal amount. In Interledger this would involve the two peers exchanging data on what their current balance is and, if there is a discrepancy, resolving this.

Reviewer: @adrianhopebailie Thoughts on this? What _does_ clearing mean in the context of Interledger?

See also: [Settlement](#settlement)                                                                       

## Connector

See instead: [Router](#router)                                                              

## Cyclic Transaction  

> A transaction that is sent out from some sender's account and is received to another one of the sender's accounts. This can be useful when rebalancing liquidity (to enable future payments), or when rebalancing stored value (to spread risk, or to take advantage of changing exchange rates).                                                                      

Reviewer: Maybe also note that a cyclic transaction can be used to trade between two different assets?

## Payment Pointer 

A payment pointer is a standardized identifier for accounts that can receive payments. It is like an email address, but for sending/receiving value.

A payment pointer resolves to an HTTPS URL that provides the location of a payment setup service endpoint at which a sender can initiate a payment to the receiver.

For example, a pointer pointer such as `$example.com/bob` will resolve to `https://example.com/bob` endpoint URL.

See also: [IL-RFC 26](https://github.com/interledger/rfcs/blob/master/0026-payment-pointers/0026-payment-pointers.md)                               

## Peer    

A router with which another router holds an account and exchanges routing information via the Interledger routing protocol.                                                                        

## Peering 

Routers "peer" (establish an account) with one another to exchange information used to determine the best route for a payment, and to route it. This is similar to how the internet service providers peer with each other (https://en.wikipedia.org/wiki/Peering).
                                                                      
## Prepare Phase   

> The first phase of an ILP payment, in which the ILP packet is passed from the sender, via the router(s), to the receiver. This represents an intent to pay, so long as the Fulfillment is delivered before the expiry.  

Reviewer: > This represents an intent to pay

Prepayment is supported, too. My preference is to describe the Prepare as an IOU between two peers, which is enforced/accounted for in the balance if the Fulfill is returned, or is called off if the Reject is returned.

                                                                      
## Receiver    

The Interledger node that is the final recipient of an Interledger payment.                                                                        

## Rejected    

> A (final) transfer state whereby funds have been returned to the sender.                                                                        

Reviewer: What does "funds" mean here?

## Route   

The path a payment has taken or a future payment could take.                                                                         

## Router  

An ILP router is similar to a network router on an IP network. Its primary function is to forward incoming packets between a sender and a receiver. 

A router may generate revenue from spreads on currency conversion, through subscription fees, or other means.
                            
## Routing Table   

A lookup table used by a router to decide who its outgoing receiver should be (the next hop, which can be another router or the destination receiver). 

## Sender  

The Interledger node that is the initiator (payer) of an Interledger payment.                                                                       

## Settlement  

The finalization of a payment, such that the payee takes the ownership of the transferred funds.                                               

> 

Reviewer: Add definition for settlement system?

SPSP uses the STREAM as transport layer protocol. 

See also: [IL-RFC 9](https://github.com/interledger/rfcs/blob/master/0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md)
                               
## STREAM  

A Transport Protocol designed to be used with ILPv4. Application Layer protocols can use STREAM to send money and data between endpoints.

See also: [IL-RFC 29](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md)                                               

## Streaming Payment   

An ongoing payment where small amounts of money are sent over time to pay for some ongoing service or rent. For instance, a user might pay for a streaming video with a streaming payment. The term "streaming payment" is commonly reserved for cases where the interval is short, from one second up to about a day.                                                                      

## Transaction 

See instead: [Payment](#payment)                                                                                                                                             

## Transport Layer 

> The transport layer transports payment data between two parties, using the [STREAM](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md) protocol. Transport Layer is the middle layer of the Interledger protocol stack, which determines the condition and encoding of the data in the ILP Packet. It also determines what details the sender and receiver need to discuss beforehand.                                                                      

Reviewer: Could we make this a little more precise?

(e.g.) The purpose of the transport layer is to ensure money and/or data sent by the sender is delivered to the recipient. This may include: splitting big payments into many smaller packets, adjusting the rate at which packets are sent, encrypting data, and providing the necessary details for the recipient to fulfill packets

# Glossary

This glossary provides definitions for key terms in the Interledger ecosystem.

## Account
   
An accounting relationship between two Interledger participants. In Interledger, two participants establish an account with one another to track the current obligations they hold with one another.                                                                         

## Address

See instead: [Interledger Address](#interledger-address)

## Application Layer

The set of protocols built upon the Transport Layer that communicate payment details between senders and receivers. 

> Application Layer protocols specify all of the data and methods required to set up a payment for a specific category of use cases or applications. For example, the Simple Payment Setup Protocol uses WebFinger and HTTPS for account detail lookup.                                                                       

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR17

## Arbitrage

> The practice of doing cyclic transactions which start and end on the same ledger, or groups of simultaneous transactions which add up to one cyclic transaction. The goal is to achieve a destination amount which is higher than the source amount by taking advantage of fluctuations in exchange rates.                                                                       

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR20

## Asset

> Something you can own that is fungible. Can be a physical item; a quantity of a physical substance; a specific physical collection of coins or banknotes; a non-physical quantity of currency; a non-physical claim to a resource; a reputation for keeping one's word; someone else's intention to obey the terms of a contract; etc. Some types of assets can be more easily traded on a ledger than others. For some asset pairs it is easier to determine a fair exchange rate than for others.                                                                       

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR24

> Something you can own. Can be a physical and unique item, a quantity of a physical substance, a specific physical collection of coins and banknotes, a non-physical quantity of currency, a non-physical claim to a resource, a reputation for keeping one's word, someone else's intention to obey the terms of a contract, etc. Some types of assets can be more easily traded on a ledger than others. Also, for some asset pairs it is easier to determine a fair exchange rate than for others.                                                                       

Reviewer 2: > It might be useful to bring in the words "fungible" and "non-fungible"

I agree with @emschwartz. While assets in general can be more broadly defined, we should probably keep our terminology limited to fungible assets. Maybe this...?
 
```suggestion
Something you can own that is fungible. Can be a physical item; a quantity of a physical substance; a specific physical collection of coins or banknotes; a non-physical quantity of currency; a non-physical claim to a resource; a reputation for keeping one's word; someone else's intention to obey the terms of a contract; etc. Some types of assets can be more easily traded on a ledger than others. For some asset pairs it is easier to determine a fair exchange rate than for others.                                                                       
```

## Balance 

> An account balance is the net total of the amounts on any packets "successfully" routed between the peers. When a node sends an ILP Prepare packet to a peer, and receives a valid ILP Fulfill response, before the expiry of the ILP Prepare, their balance with that peer decreases. When a node receives an ILP Prepare packet and returns a valid ILP Fulfill response, before the expiry of the ILP Prepare, their balance with that peer increases.                                                                        

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR28

## Bandwidth

> The limit, set by a router, to the total value of ILP packets an account can send during a given time period. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's bandwidth with its peers.                                                                       

Author: I'm a bit confused.  What's the difference between throughput and bandwidth?

> The limit, set by a router, to the total value of ILP packets an account can send during a given time period. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's bandwidth with its peers.                                                                       

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR32

Fixed at other places as well.

> The limit, set by a router, to the total value of ILP packets an account can send during a given time period. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's bandwidth with its peers.                                                                       

Reviewer 2: Great discussion. It seems like we should stop using the term "bandwidth" to describe `max-in-flight`. Reason being is that bandwidth is typically measured relative to some frequency. E.g., Internet Pipe Bandwidth: 5gb-per-second; Water Pipe Bandwidth: 3 liters-per-second; RF Antenna: 30 GHz; etc. 

However, "max in flight" doesn't necessarily correlate to a frequency or period. If my `max-in-flight` is 5, then it doesn't matter if I have 5 units in-flight for 1 minute or 1 hour.

If this is correct, then I would expect something like the following definitions to hold:

* **Account Max In-Flight**: The maximum value an account can have in-flight at any given moment in time (e.g., $5). 
* **Account Bandwidth**: The maximum value that can be streamed per-second ($5 / second).
* **Account Throughput**: The current value being transferred over a given time period. (e.g., Your _account bandwidth_ is currently $5/second, but your _account throughput_ is only $1/second because maybe the receiver can only receive that much per second for some reason).

All that to say, maybe the definition in the current PR should just say this:

```suggestion
The limit, set by a router, of the total value an account can have pending, or "in-fligh" at any given moment in time. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's liquidity with its peers.
```

> The limit, set by a router, to the total value of ILP packets an account can send during a given time period. The limit may either be based on how much the router trusts the account-holder, or it may be used to prevent one account from tying up all of the router's bandwidth with its peers.                                                                       

Author: Updated the definition with suggested text by @sappenin 
https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR33

## Clearing    

> The transferring (transmitting, reconciling, and confirming) of funds from payer to payee, such that the payer's account is debited for the promised amount and the payee's account is credited for the equal amount. 

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR36

See also: [Settlement](#settlement)                                                                       

## Connector

See instead: [Router](#router)                                                              

## Cyclic Transaction  

> A transaction where the destination account is the same account on the ledger as the source account. This can be useful when rebalancing liquidity (to enable future payments), or when rebalancing stored value (to spread risk, or to take advantage of changing exchange rates).                                                                      

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR46

> A transaction where the destination account is the same account on the ledger as the source account. This can be useful when rebalancing liquidity (to enable future payments), or when rebalancing stored value (to spread risk, or to take advantage of changing exchange rates).                                                                      

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR46

## Destination Account 

The account of the receiver whose address is included in the Interledger packet.                                                                       

## Destination Amount  

> The amount to be received by the receiver through [ILP](#interledger-protocol-ilp).                                                                      

Author: Isn't this the amount that the sender puts in a Prepare packet?

> The amount to be received by the receiver through [ILP](#interledger-protocol-ilp).                                                                      

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR53

## Distributed Ledger

A ledger that is operated by a group of entities and runs on multiple servers. This term is used somewhat interchangeably with "Blockchain" or "Decentralized Ledger".                                                                        

## Exchange Rate

The price of one ledger's asset in terms of another ledger's asset. Routers may generate revenue from the difference in value between incoming and outgoing transfers. The exchange rate between a sender and a receiver is determined by the product of exchange rates at each hop.                                                                         

## Fulfillment 

> Also known as the "preimage" or "hash preimage", a fulfillment is  a 32-byte value used to trigger the execution of a transfer. In most Interledger payments, the fulfillment is known only to the receiver (or in the case of the STREAM protocol it is known to the sender and the receiver).                                                                        

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR69

## Interledger (in "Let's use Interledger for that!")  

The Interledger protocol stack.                                                                         

## interledger (in "An interledger network")

A network of two or more ledgers that use the Interledger protocol stack. 

See also: [Interledger (The)](#interledger-in-the-interledger) 

The adjective 'interledger' is also sometimes used to refer to transactions that cross multiple ledgers, even if ILP is not used (similar to 'international' meaning 'across multiple nations').                                                                         

## Interledger (in "The Interledger")  

The public network of ledgers connected via the Interledger protocol stack.                                                                         

## Interledger Address 

An Interledger address provides a way to route ILP packets to their intended destination through a series of hops, including any number of ILP routers. 

> An Interledger address is a dot-separated string that contains prefixes to group ledgers. An example address might look like `g.us.acmebank.acmecorp.sales.199` or `g.crypto.bitcoin.1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2`. 

Reviewer 3: this is no longer really the case in Interledger. Because the network is now built with connectors and bilateral accounts rather than ledgers and ledger accounts, ILP addresses are now built up of segments which represent connectors. Adding more segments on the end typically means you're referring to a connector's peer with relationship child

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR99

> Note that anyone can claim to have a certain Interledger address, there is no registry of them. Whether or not a payment ends up at the intended receiver is ultimately safeguarded by the hashlock condition, not by enforcement of address ownership. 

Reviewer 3: there's ongoing discussion of a secure routing protocol which would actually allow address ownership. Not sure where active discussion on this is happening so I don't know where we can link it to

Reviewer 2: I brought up secure routing during the ILP call yesterday - consensus on the call was to form a working group and work on secure routing as a "slow burn" since it's not urgent while the network is very nascent. Until we have some sort of PKI in Interledger, I think for now we can move forward using the current model of ILP address allocation.

See also: [IL-RFC 15](https://github.com/interledger/rfcs/blob/master/0015-ilp-addresses/0015-ilp-addresses.md)                                                                      

## Interledger Architecture    

The Interledger protocol stack and the design principles behind it.

See also: [IL-RFC 1](https://github.com/interledger/rfcs/blob/master/0001-interledger-architecture/0001-interledger-architecture.md)                                                                       

## Interledger Layer   

> The second layer of the Interledger protocol stack, consisting of the Interledger Protocol version 4 (ILPv4). This layer is concerned with currency amounts, routing, and whether each step in a payment arrives in time or expires. This protocol finds a path to connect a sender and receiver using any number of intermediaries.                                                                     

Reviewer 3: Are we still counting the ledger as layer 1? If so ILP would be layer 3

Author: Looking at this diagram, I am assuming that the grey portion comprises the Interledger layers. That's why I have mentioned ILP as the second layer. Is that diagram outdated?

Reviewer 2: I think what @sharafian is referencing is the protocol layering outlined in [this blog post](https://medium.com/xpring/layer-3-is-for-interoperability-ca387fa5f7e2) by @emschwartz. 

In that post, Interledger protocols are layer-3 and "up" (i.e., layers 4 and 5 too). I think we should maintain _that_ layering because it accounts for related technologies that aren't Interledger per-se, like Bitcoin, which would be a Layer-1 ledger.

Author: I have removed the numbering from this definition to avoid confusion.
https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR117

> The second layer of the Interledger protocol stack, consisting of the Interledger Protocol version 4 (ILPv4). This layer is concerned with currency amounts, routing, and whether each step in a payment arrives in time or expires. This protocol finds a path to connect a sender and receiver using any number of intermediaries.                                                                     

Reviewer 3: this protocol finds a path -> this protocol layer finds a path

Technically routing is not included the the ILPv4 protocol, there's a separate ILP routing protocol and connector-connector protocol for routing information

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR110

See also: [IL-RFC 1](https://github.com/interledger/rfcs/blob/master/0001-interledger-architecture/0001-interledger-architecture.md#interledger-protocol)

## Interledger Packet  

Also known as the ILP Packet.

This is the binary data packet that is routed across all hops from sender to receiver in all Interledger payments. ILPv4 has three packet types: Prepare, Fulfill, and Reject, which roughly correspond to request, response, and error messages.

The ILP Prepare packet contains the details of the Interledger transaction, including the destination ILP Address, the amount to be sent, an expiration timestamp, a Condition, and data for the receiver. The ILP Fulfill packet contains the Fulfillment and data from the receiver for the sender. The ILP Reject packet contains an error code and message, as well as the ILP Address of the node that produced the error.                                                                                                                                 

> The packet includes metadata which the receiver may need before fulfilling the payment. It also includes metadata (such as destination account and amount) that routers use to determine what path to take. Alternatively, if the incoming amount is not large enough, compared to the destination amount in the ILP packet, the router may choose not to route the payment at all.

Reviewer 3: this makes it sound like these fields are different from the ones mentioned in the previous paragraph

## Asset

Something you can own. Can be a physical and unique item, a quantity of a physical substance, a specific physical collection of coins and banknotes, a non-physical quantity of currency, a non-physical claim to a resource, a reputation for keeping one's word, someone else's intention to obey the terms of a contract, etc. Some types of assets can be more easily traded on a ledger than others. Also, for some asset pairs it is easier to determine a fair exchange rate than for others.                                                                       

## Balance 

An account balance is the net total of the amounts on any packets "successfully" routed between the peers. When a node sends an ILP Prepare packet to a peer, and receives a valid ILP Fulfill response, before the expiry of the ILP Prepare, their balance with that peer decreases.When a node receives an ILP Prepare packet and returns a valid ILP Fulfill response, before the expiry of the ILP Prepare, their balance with that peer increases.                                                                        

> ## Bandwidth

Author: Added both terms separately and provided link to Bandwidth definition.

## Clearing    

The transferring (transmitting, reconciling, and confirming) of funds from payer to payee, such that the payer's account is debited for the promised amount and the payee's account is credited for the equal amount. 

See also: [Settlement](#settlement)                                                                       

## Connector

See instead: [Router](#router)                                                              

## Cyclic Transaction  

A transaction where the destination account is the same account, on the same ledger, as the source account. This can be useful when rebalancing liquidity (to enable future payments), or when rebalancing stored value (to spread risk, or to take advantage of changing exchange rates).                                                                      

## Destination Account 

> The account of the receiver whose address is included in the Interledger packet. Note that anyone can claim to have a certain Interledger address; address ownership is not enforced.                                                                       

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR100

## Destination Amount  

The amount to be received by the receiver.                                                                      

## Distributed Ledger

A ledger that is operated by a group of entities and runs on multiple servers. This term is used somewhat interchangeably with "Blockchain" or "Decentralized Ledger".                                                                        

## Exchange Rate

> The price of one ledger's asset in terms of another ledger's asset. Routers may generate revenue from the difference in value between incoming and outgoing transfers. The exchange rate between source and destination is determined by the product of exchange rates at each hop.                                                                         

Author: Fixed.

> ## Fulfillment 

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR184

A 32-byte value used to trigger the execution of a transfer. In most Interledger payments, the fulfillment is known only to the receiver (or in the case of the STREAM protocol it is known to the sender and the receiver).                                                                        

## Hold    

Part of the sender's, router's, and/or receiver's account balance is put "on hold" to reserve that balance for a specific ILP payment and is temporarily unavailable for use in other payments. The money is on hold while the ILP transfer is in the prepared state. If the transfer is executed, the money is no longer on hold, and added to the balance of the receiver of the transfer. If the transfer is rejected, the money is no longer on hold either, but is instead returned to the balance of the sender of the transfer.                                                                      

## Hop 

> The movement of an ILP packet from one router to another as it travels through the Interledger network to reach its destination.                                                                         

Reviewer 2: Maybe we define this as the connection between two nodes, perhaps something like this:

`A colloquial term for the relationship between two peers that enables value to move. ILP packets travel from one router "hop" to another as they make their way through the Interledger network.`

## Interledger Address 

An Interledger address provides a way to route ILP packets to their intended destination through a series of hops, including any number of ILP routers. 

An Interledger address is a dot-separated string that contains prefixes to group ledgers. An example address might look like `g.us.acmebank.acmecorp.sales.199` or `g.crypto.bitcoin.1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2`. 
Note that anyone can claim to have a certain Interledger address, there is no registry of them. Whether or not a payment ends up at the intended receiver is ultimately safeguarded by the hashlock condition, not by enforcement of address ownership. 

See also: [IL-RFC 15](https://github.com/interledger/rfcs/blob/master/0015-ilp-addresses/0015-ilp-addresses.md)                                                                      

## Interledger Architecture    

The Interledger protocol stack and the design principles behind it.

See also: [IL-RFC 1](https://github.com/interledger/rfcs/blob/master/0001-interledger-architecture/0001-interledger-architecture.md)                                                                       

## Interledger Layer   

> The second layer of the Interledger protocol stack, consisting of the Interledger Protocol version 4 (ILPv4). This layer is concerned with currency amounts, routing, and whether each step in a payment arrives in time or expires. This protocol finds a path to connect a sender and receiver using any number of intermediaries.                                                                        

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR114

## Interledger Protocol (ILP)  

The set of rules that define how Interledger nodes should communicate with each other. 

Colloquially, the whole Interledger stack is sometimes referred to as ILP. Technically, however, the Interledger Protocol is only one layer in the Interledger protocol stack.

See also: [IL-RFC 27](https://github.com/interledger/rfcs/blob/master/0027-interledger-protocol-4/0027-interledger-protocol-4.md)                                                             

## Interledger Protocol Stack  

The stack consisting of the link layer, Interledger layer, a choice of transport layer protocols, and the application layer.

## ILPv4

See instead: [Interledger Protocol](#interledger-protocol-ilp)                                                                    

## Intermediate (transfer, ledger, amount) 

Any transfer/ledger/amount that is neither directly adjacent to the sender nor the receiver.                                                                        

## Ledger  

> Stateful systems that are used for clearing, settlement, and tracking the ownership of assets. Ledgers contain buckets of assets known as accounts and record transfers between them. Each account has a balance, which is the amount of the ledger's assets the account holds. Account balances may be positive or negative, representing assets or liabilities.                                                                       

Author: Removed the word "clearing".

## Payment 

In the context of ILPv4, a payment is understood to mean the transfer of value from the sender (payer) to the receiver (payee). Higher-level protocols may execute a "payment" by sending a series of ILP Packets whose sum is equal to the desired payment value.                                                                      

## Payment Channel 

A payment channel is a medium used by two parties to perform multiple value transactions amongst them, without sending these transactions to the underlying ledger. One or both parties will generally transfer some of their value assets on the ledger to the payment channel. Value transfers are then made between the two parties by exchanging "claims" that update their balance. When one of the parties wishes to close the channel, they submit the final balance to the ledger.                                                                      

## Payment Pointer 

> A payment pointer is a standardized identifier for accounts that can receive payments. It is like an email address, but for sending/receiving value.

Author: https://github.com/interledger/interledger.github.io/pull/123/files#diff-0f4b6322d1459dbe5c7f5eb93d2cd2efR184

A payment pointer resolves to an HTTPS URL that provides the location of a payment setup service endpoint at which a sender can initiate a payment to the receiver.                                                                  

## Peer    

A router with which another router holds an account.                                                                        

## Peering 

Routers "peer" (establish an account) with one another to exchange information used to determine the best route for a payment, and to route it. This is similar to how internet service providers peer with each other (https://en.wikipedia.org/wiki/Peering).                                                                         

## Prepare Phase   

> The first phase of an ILP payment, in which the ILP packet is passed from the sender, via the router(s), to the receiver. Also, money is put on hold during this phase.                                                                         

Reviewer 4: ```suggestion
The first phase of an ILP payment, in which the ILP packet is passed from the sender, via the router(s), to the receiver. This represents an intent to pay, so long as the Fulfillment is delivered before the expiry.                                                                   
```

> ## Prepared    

Reviewer 4: Remove

The state of a transfer in which the source account's funds are put on hold by the ledger until either the expiry is reached or the condition is fulfilled.                                                                         

## Receiver    

The Interledger node that is the final recipient of an Interledger payment.                                                                        

## Rejected    

A (final) transfer state whereby funds have been returned to the sender.                                                                        

## Route   

> A set of transfers chained together by the routers between them. The route is the path a payment has taken, or a future payment could take.                                                                         

Reviewer 4: Thinking of it as a chain of transfers was the ILPv1 way to think of it

## Routing Table   

A lookup table used by a router to decide who its outgoing receiver should be (the next hop, which can be another router or the destination receiver).                                                                         

## Sender  

The Interledger node that is the initiator (payer) of an Interledger payment.                                                                       

## Settlement  

The finalization of a payment, such that the payee takes the ownership of the transferred funds.                                                                        

## Simple Payment Setup Protocol (SPSP)

An application layer protocol for exchanging payment information - destination account and secret - over HTTPS, between a payee and a payer, to facilitate payment over Interledger. 

SPSP uses the STREAM as transport layer protocol. 

See also: [IL-RFC 9](https://github.com/interledger/rfcs/blob/master/0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md)                                                                        

> ## Source (transfer, ledger, amount)   

Reviewer 4: Remove

The transfer/ledger/amount directly adjacent to the sender.                                                                         

## STREAM  

A Transport Protocol designed to be used with ILPv4. Application Layer protocols can use STREAM to send money and data between endpoints.

See also: [IL-RFC 29](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md)                                                                   

## Streaming Payment   

An ongoing payment where small amounts of money are sent over time to pay for some ongoing service or rent. For instance, a user might pay for a streaming video with a streaming payment. The term "streaming payment" is commonly reserved for cases where the interval is short, from one second up to about a day.                                                                      

## Transaction 

See instead: [Payment](#payment)                                                                     

> ## Transfer    

Reviewer 4: Remove

The movement of assets from one peer to another. Multiple transfers can be chained together into one multi-hop payment.                                                                         

## Transport Layer 

The transport layer transports payment data between two parties, using the [STREAM](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md) protocol. Transport Layer is the middle layer of the Interledger protocol stack, which determines the condition and encoding of the data in the ILP Packet. It also determines what details the sender and receiver need to discuss beforehand.                                                                      

## Value   

> A financial asset that represents currency. It can be fiat (such as USD, EUR, JPY, etc.) or digital (such as  BTC, ETH, XRP, etc.). 

Reviewer 4: I think currency is a subset of "value". Value is harder to define but can also include non-currency assets

Author: This definition is with-respect-to  Interledger. I agree that value can include non-currency assets. But currently, I think, in Interledger we can only transfer currency. No?

Reviewer 2: > in Interledger we can only transfer currency. No?

Technically Interledger can transfer any fungible asset. So, imagine: "kernels of corn" or "shares of apple stock" or whatever.

## Cyclic Transaction  

A transaction where the destination account is the same account on the ledger as the source account. This can be useful when rebalancing liquidity (to enable future payments), or when rebalancing stored value (to spread risk, or to take advantage of changing exchange rates).                                                                      

## Destination Account 

The account of the receiver whose address is included in the Interledger packet. Note that anyone can claim to have a certain Interledger address; address ownership is not enforced.                                                                       

## Destination Amount  

> The amount to be received by the receiver through ILP.                                                                      

Reviewer 2: Since ILP is itself a term, and we haven't introduce that term, we should link to it from here.

Author: Fixed.

## Distributed Ledger

A ledger that is operated by a group of entities and runs on multiple servers. This term is used somewhat interchangeably with "Blockchain" or "Decentralized Ledger".                                                                        

## Exchange Rate

The price of one ledger's asset in terms of another ledger's asset. Routers may generate revenue from the difference in value between incoming and outgoing transfers. The exchange rate between a sender and a receiver is determined by the product of exchange rates at each hop.                                                                         

## Fulfillment 

> Also known as the "preimage" or "hash preimage", fulfillment is  a 32-byte value used to trigger the execution of a transfer. In most Interledger payments, the fulfillment is known only to the receiver (or in the case of the STREAM protocol it is known to the sender and the receiver).                                                                        

Reviewer 2: Missing an `a` before "fulfillment":

..."hash preimage", **a** fulfillment is...

Author: Fixed.
